backend/src/apiserver/server/job_server.go

```diff
@@ -90,22 +89,19 @@ func (s *JobServer) createJob(ctx context.Context, job *model.Job) (*model.Job,
 	if err != nil {
 		return nil, util.Wrapf(err, "Failed to create a run due to invalid experimentId and namespace combination")
 	}
+	if common.IsMultiUserMode() && namespace == "" {
+		return nil, util.NewInvalidInputError("Recurring run cannot have an empty namespace in multi-user mode")
+	}
 	job.ExperimentId = experimentId
 	job.Namespace = namespace
-	// Validate authorization
-	if common.IsMultiUserMode() {
-		if job.Namespace == "" {
-			return nil, util.NewInternalServerError(util.NewInvalidInputError("Recurring run cannot have an empty namespace in multi-user mode"), "Failed to create a recurring run due to empty namespace")
-		}
-		resourceAttributes := &authorizationv1.ResourceAttributes{
-			Namespace: job.Namespace,
-			Verb:      common.RbacResourceVerbCreate,
-			Resource:  common.RbacResourceTypeJobs,
```

Comment:
Is it intentional that `Resource` is deleted from the new code?

Reply:
Yes. The resource, group, and version are added in `canAccessJob`.

Reply:
I see, so probably a cleaner way is to always construct the full `ResourceAttributes` object, and reduce all the `canAccessXXX` functions into `canAccessResource`, WDTY? This can be out-of-scope.